flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -782,16 +797,33 @@ private boolean isUnboundedSource(final Transformation<?> transformation) {
 		for (int i = 0; i < allInputIds.size(); i++) {
 			Collection<Integer> inputIds = allInputIds.get(i);
 			for (Integer inputId: inputIds) {
-				streamGraph.addEdge(inputId,
-					transform.getId(),
-					i + 1
-				);
+				streamGraph.addEdge(inputId, transform.getId(), i + 1);
 			}
 		}
 
 		return Collections.singleton(transform.getId());
 	}
 
+	/**
+	 * Returns a list of lists containing the ids of the nodes in the transformation graph
+	 * that correspond to the provided transformations. Each transformation may have multiple nodes.
```

Comment:
Maybe add here that `Transformations` will be translated if they are not already translated.